File: EventBa.Model/Helpers/Meta.cs
namespace EventBa.Model.Helpers;

public class Meta
{
    public int? TotalCount { get; }
    public int PageNumber { get; }
    public int PageSize { get; }

    public int TotalPages => (int)Math.Ceiling((double)TotalCount / PageSize);
    public bool HasPrevious => PageNumber > 1;
    public bool HasNext => PageNumber < TotalPages;

    public Meta(int totalCount, int pageNumber, int pageSize)
    {
        TotalCount = totalCount;
        PageNumber = pageNumber;
        PageSize = pageSize;
    }
}

File: UI/eventba_admin/lib/models/meta/meta.dart
import 'package:json_annotation/json_annotation.dart';

part 'meta.g.dart';

@JsonSerializable()
class Meta {
  @JsonKey(name: 'totalCount')
  final int? totalCount;

  @JsonKey(name: 'pageNumber')
  final int pageNumber;

  @JsonKey(name: 'pageSize')
  final int pageSize;

  @JsonKey(name: 'totalPages')
  final int totalPages;

  @JsonKey(name: 'hasPrevious')
  final bool hasPrevious;

  @JsonKey(name: 'hasNext')
  final bool hasNext;

  Meta({
    this.totalCount,
    required this.pageNumber,
    required this.pageSize,
    required this.totalPages,
    required this.hasPrevious,
    required this.hasNext,
  });

  factory Meta.fromJson(Map<String, dynamic> json) => _$MetaFromJson(json);
  Map<String, dynamic> toJson() => _$MetaToJson(this);
}
